src/connectors/impls/gbq/writer/sink.rs

```diff
@@ -0,0 +1,443 @@
+                result,
+            );
+        }
+        TableType::Json => {
+            warn!("Found a field of type JSON, this is not supported, ignoring.");
+        }
+        TableType::Interval => {
+            warn!("Found a field of type Interval, this is not supported, ignoring.");
+        }
+
+        TableType::Unspecified => {
+            warn!("Found a field of unspecified type - ignoring.");
+        }
+    }
+
+    Ok(())
+}
+
+impl JsonToProtobufMapping {
+    pub fn new(vec: &Vec<TableFieldSchema>) -> Self {
+        let descriptor = map_field("table", vec);
+
+        Self {
+            descriptor: descriptor.0,
+            fields: descriptor.1,
+        }
+    }
+
+    pub fn map(&self, value: &Value) -> Result<Vec<u8>> {
+        let mut result = vec![];
+        if let Some(obj) = value.as_object() {
+            for (key, val) in obj {
+                if let Some(field) = self.fields.get(&key.to_string()) {
+                    encode_field(val, field, &mut result)?;
+                }
+            }
+        }
+
+        Ok(result)
+    }
+
+    pub fn descriptor(&self) -> &DescriptorProto {
+        &self.descriptor
+    }
+}
+impl GbqSink {
+    pub async fn new(config: Config) -> Result<Self> {
+        Ok(Self {
+            client: None,
+            write_stream: None,
+            mapping: None,
+            config,
+        })
+    }
+}
+
+#[async_trait::async_trait]
+impl Sink for GbqSink {
+    async fn on_event(
+        &mut self,
+        _input: &str,
+        event: Event,
+        ctx: &SinkContext,
+        _serializer: &mut EventSerializer,
+        _start: u64,
+    ) -> Result<SinkReply> {
+        let client = self
+            .client
+            .as_mut()
+            .ok_or(ErrorKind::BigQueryClientNotAvailable(
+                "The client is not connected",
+            ))?;
+        let write_stream =
+            self.write_stream
+                .as_ref()
+                .ok_or(ErrorKind::BigQueryClientNotAvailable(
+                    "The write stream is not available",
+                ))?;
+        let mapping = self
+            .mapping
+            .as_ref()
+            .ok_or(ErrorKind::BigQueryClientNotAvailable(
+                "The mapping is not available",
+            ))?;
+
+        let request = AppendRowsRequest {
+            write_stream: write_stream.name.clone(),
+            offset: None,
+            trace_id: "".to_string(),
+            rows: Some(append_rows_request::Rows::ProtoRows(ProtoData {
+                writer_schema: Some(ProtoSchema {
+                    proto_descriptor: Some(mapping.descriptor().clone()),
+                }),
+                rows: Some(ProtoRows {
+                    serialized_rows: vec![mapping.map(event.data.parts().0)?],
+                }),
+            })),
+        };
+
+        let append_response = client
+            .append_rows(stream::iter(vec![request]))
+            .timeout(Duration::from_secs(10))
```

Comment:
hardcoding a timeout here seems to be risky. Would it be possible to omit it or make it configurable to omit / or select how long it is.